services/event/models/event.go

```diff
@@ -1,15 +1,17 @@
 package models
 
 type Event struct {
-	ID          string          `json:"id"                  validate:"required"`
-	Name        string          `json:"name"                validate:"required"`
-	Description string          `json:"description"         validate:"required"`
-	StartTime   int64           `json:"startTime"           validate:"required"`
-	EndTime     int64           `json:"endTime"             validate:"required"`
-	Locations   []EventLocation `json:"locations"           validate:"required,dive,required"`
-	Sponsor     string          `json:"sponsor"`
-	EventType   string          `json:"eventType"           validate:"required,oneof=MEAL SPEAKER WORKSHOP MINIEVENT QNA OTHER"`
-	Points      int             `json:"points"`
+	ID                 string          `json:"id"                  validate:"required"`
+	Name               string          `json:"name"                validate:"required"`
+	Description        string          `json:"description"         validate:"required"`
+	StartTime          int64           `json:"startTime"           validate:"required"`
+	EndTime            int64           `json:"endTime"             validate:"required"`
+	Locations          []EventLocation `json:"locations"           validate:"required,dive,required"`
+	Sponsor            string          `json:"sponsor"`
+	EventType          string          `json:"eventType"           validate:"required,oneof=MEAL SPEAKER WORKSHOP MINIEVENT QNA OTHER"`
+	InPersonPoints     int             `json:"inPersonPoints"`
+	InPersonVirtPoints int             `json:"inPersonVirtPoints"`
+	VirtualPoints      int             `json:"virtualPoints"`
```

Comment:
Nit: make all (or one, and add defaults) of these a required field too?

Reply:
It appears that the `required` tag checks if the parameter is not zero if the data type is a number. Also, I'm not entirely too sure if there is a value that will always be set. We could have events that are in-person exclusive, virtual-only, or a combination of the two, so not all point values are guaranteed will have a value greater than 0.

Reply:
Update: we could do something like:
```suggestion
        InPersonPoints     int             `json:"inPersonPoints"      validate:"required_without=VirtualPoints"`
	InPersonVirtPoints int             `json:"inPersonVirtPoints"  validate:"required_with_all=InPersonPoints VirtualPoints"`
	VirtualPoints      int             `json:"virtualPoints"       validate:"required_without=InPersonPoints"`
}
```

Only issue that validator.v9 is giving some issues with it:
```bash
2022/01/03 03:04:18 http: panic serving 127.0.0.1:48154: Undefined validation function 'required_without' on field 'InPersonPoints'
```

Fortunately, validator.v10 worked for me.

Reply:
Sounds good. To clarify, are we assuming an event with no `InPersonPoints` or `VirtualPoints` will correspond to 0 points being awarded for participating in the event in person and virtually respectively?

Reply:
Yes, that's what will happen. Something to note is that some events are not meant to have points awarded to users since the event could be representing food/meals for example.

So, for right now, I'll just leave it as is, but if we update to validator/v10, the following tags should be good:
```golang
	InPerson     uint            `json:"inPerson"`
	InPersonVirt uint            `json:"inPersonVirt"  validate:"required_with_all=InPerson Virtual"`
	Virtual      uint            `json:"virtual"`
```

Reply:
Yup, makes sense. Having 0 as the default value is reasonable, and shouldn't be too disastrous if the caller forgets to set certain point values.